src/idl_gen_cpp.cpp

```diff
@@ -1304,6 +1308,11 @@ class CppGenerator : public BaseGenerator {
                   Name(field) + "(" + native_default->constant + ")";
             }
           }
+        } else if(field.value.type.base_type == BASE_TYPE_VECTOR) {
```

Comment:
Actually, I don't think this part is needed, as vectors are initialised just fine without a call to their empty ctor.
Also, this makes a class' initializer list longer, which is not intended.